modules/solid_properties/src/materials/ThermalSolidPropertiesMaterial.C

```diff
@@ -0,0 +1,51 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ThermalSolidPropertiesMaterial.h"
+
+InputParameters
+ThermalSolidPropertiesMaterial::validParams()
+{
+  InputParameters params = SolidPropertiesMaterial::validParams();
+  params.addClassDescription("Material providing solid thermal properties");
+  params.addRequiredCoupledVar("temperature", "Temperature");
+  params.addParam<std::string>(
+      "cp_name", "cp_solid", "Name to be used for the isobaric specific heat");
+  params.addParam<std::string>("k_name", "k_solid", "Name to be used for the thermal conductivity");
+  params.addParam<std::string>("rho_name", "rho_solid", "Name to be used for the density");
```

Comment:
use NS namespace? Or create a new name space and include it in NS?

Reply:
Probably heat conduction namespace would be more appropriate. As far as I can tell, it doesn't have these quantities named anywhere yet, but they should go in `HeatConductionNames.h`. It uses `density`, `specific_heat`, and `thermal_conductivity` as default names in its objects, so those are probably the names that should be used.

Reply:
See comment below - I created a namespace in the solid properties app that we can update other modules to use in a separate PR.